src/iop/borders.c

```diff
@@ -373,24 +430,55 @@ void modify_roi_out(struct dt_iop_module_t *self, struct dt_dev_pixelpipe_iop_t
   const float size = fabsf(d->size);
   if(size == 0) return;
 
-  if(d->aspect == DT_IOP_BORDERS_ASPECT_CONSTANT_VALUE)
+  const gboolean is_constant_border = d->aspect == DT_IOP_BORDERS_ASPECT_CONSTANT_VALUE;
+
+  dt_iop_basis_t basis = d->basis;
+  if (is_constant_border && !d->max_border_size)
```

Comment:
And if the migration is properly done in `legacy_params` we can probably remove this part.